Create Websocket 2.1 API bundles

volosied opened the pull request:
Part of #18745 and #20588

- [x] - Created two separate bundles due to the API reorganization. Two bundle were created so that I could list maven coordinates for each API in the private feature file. 
 - io.openliberty.jakarta.websocket.2.1
 - io.openliberty.jakarta.websocket.client.2.1

- [x] - Enabled our websocket FAT to run EE10 features by default in lite mode (for Java 11+).

- [x] - Updated beans.xml to include the version to avoid error. The version is a requirement for CDI 4.0. 
```
[5/20/22, 0:42:56:289 EDT] 00000025 com.ibm.ws.app.manager.AppMessageHelper                      E CWWKZ0002E: An exception occurred while starting the application cdi. The exception message was: com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.DeploymentException: WELD-001408: Unsatisfied dependencies for type StringBuffer with qualifiers @Named
  at injection point [BackedAnnotatedField] @Inject @Named public cdi.war.AnnotatedCDIIdleTimeoutServerEP.N2
  at cdi.war.AnnotatedCDIIdleTimeoutServerEP.N2(AnnotatedCDIIdleTimeoutServerEP.java:0)
  ```
- [x] - Added sources to API bundles

- [x] - Update copyrights and comments 

- [x] - Created new FAT: io.openliberty.wsoc.21.internal_fat. It contains 4 tests which verify no timeout is set when 0 or a negative time is set.

volosied commented:
#build

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/bnd.bnd:
```diff
@@ -0,0 +1,38 @@
+    com.ibm.tx.jta;version=latest,\
+    com.ibm.ws.tx.embeddable;version=latest,\
+    com.ibm.ws.transaction.test.util;version=latest,\
+    com.ibm.ws.org.slf4j.api;version=latest, \
```
new line please

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/FATSuite.java:
```diff
@@ -0,0 +1,48 @@
+
+/**
+ * Collection of all example tests
+ */
```
Update comment please.

volosied replied:
Updated

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/common/Constants.java:
```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2014 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
```
2014?

pnicolucci replied:
Are these classes just brought over from an existing component? That would explain the 2014 copyright? If that's the case I think it makes sense to keep them since we're just moving the class into a new component.

volosied replied:
Correct. Many classes were ported from the existing FAT

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java:
```diff
@@ -0,0 +1,216 @@
+        assertNotNull("Timeout message not found!", result);
+    }
+
+}
```
new line

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/tests/all/TimeOutTest.java:
```diff
@@ -0,0 +1,62 @@
+        wsocTest.runEchoTest(new TimeOutClientEP.TimeOutTest(input1), uri, output1);
+
+    }
+}
```
new line

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/publish/servers/basicTestServer/bootstrap.properties:
```diff
@@ -0,0 +1,11 @@
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+bootstrap.include=../testports.properties
```
new line

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/test-applications/basic.war/resources/WEB-INF/web.xml:
```diff
@@ -0,0 +1,27 @@
+        <url-pattern>/SingleRequest</url-pattern>
+    </servlet-mapping>
+	
+</web-app>
```
new line

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/test-applications/basic.war/resources/index.html:
```diff
@@ -0,0 +1,378 @@
+           </div>
+
+    </body>
+</html>
```
new line

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/test-applications/basic.war/src/basic/war/ZeroTimeOutServerEP.java:
```diff
@@ -0,0 +1,46 @@
+        return String.valueOf(this.session.getMaxIdleTimeout());
+    }
+
+}
```
new line

pnicolucci commented on dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/SessionIdleTimeout.java:
```diff
@@ -36,6 +36,7 @@ public class SessionIdleTimeout implements Runnable {
     public SessionIdleTimeout(String sessionID, long maxIdleTimeout, WsocConnLink wsocLink) {
         this.sessionID = sessionID;
         this.maxIdleTimeout = maxIdleTimeout;
+        Thread.dumpStack();
```
I assume this was here for some debugging and needs to get cleaned up?

volosied replied:
Yes, I definitely need to remove that.

pnicolucci commented on dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/SessionIdleTimeout.java:
```diff
@@ -44,7 +45,8 @@ public void restartIdleTimeout(long maxIdleTimeout) {
         if (maxIdleTimeout < 1) {
             if (tc.isDebugEnabled()) {
-                Tr.debug(tc, "Session idle timeout is default which is no time out");
+                System.out.println("debug idle");
```
Also for some debugging during dev that should be removed?

pnicolucci commented on dev/io.openliberty.jakarta.websocket.2.1/bnd.bnd:
```diff
@@ -0,0 +1,34 @@
+publish.wlp.jar.suffix: dev/api/spec
+
+-buildpath: \
+	jakarta.websocket:jakarta.websocket-api;version=2.1.0
```
Is there a module-info that we need to exclude?

volosied replied:
No need as it's not included in the first place.

pnicolucci commented on dev/io.openliberty.jakarta.websocket.client.2.1/bnd.bnd:
```diff
@@ -0,0 +1,34 @@
+publish.wlp.jar.suffix: dev/api/spec
+
+-buildpath: \
+	jakarta.websocket:jakarta.websocket-client-api;version=2.1.0
```
Is there a module-info that we need to exclude?

volosied replied:
Not required as it's _not_ added by default.  If I was including resource files then it might have to be added.

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/FATSuite.java:
```diff
@@ -0,0 +1,48 @@
+ * below should represent all of the test cases for this FAT.
+ */
+@SuiteClasses({
+                BasicTest.class
```
This test is also run in the existing wsoc FAT isn't it? Is the plan just to keep this here until we have some 2.1 specific tests and then remove it?

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/endpoints/client/basic/TimeOutClientEP.java:
```diff
@@ -0,0 +1,100 @@
+
+/**
+ *
+ */
```
Can you add some information here? Is this a new test class for WebSocket 2.1?

volosied replied:
Added a small comment for the class

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java:
```diff
@@ -0,0 +1,216 @@
+// not deleted so the same pattern would not be repeated.
+
+/**
+ * Tests WebSocket Stuff
```
Is this a new websocket 2.1 test class or an existing class from an older websocket FAT?

volosied replied:
It's a new test class. I renamed it to Basic21Test instead

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/fat/src/io/openliberty/wsoc/tests/all/TimeOutTest.java:
```diff
@@ -0,0 +1,62 @@
+import junit.framework.Assert;
+
+/**
+ * Tests WebSocket Stuff
```
New or existing test for websocket 2.1? Should this be in the FATSuite?

volosied replied:
New Test for 2.1. It should _not_ be in the FatSuite. Basic21Test calls into this TimeOutTest.  

This is just the way it was done in the original FAT, not sure why.  The likely answer is due to the testSSC* test cases where we use the WAS implementation for both client and server. The "normal" tests use Jetty's wsoc impl as the client.  

TimeOutTest is packaged into the basic21.war and can be invoked via the URL (as part of the testSSC* test).

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/test-applications/basic.war/src/basic/war/NegativeTimeOutServerEP.java:
```diff
@@ -0,0 +1,48 @@
+
+/**
+ *
+ */
```
Can you add some doc?

pnicolucci replied:
Would be nice in these new classes to have a bit of an overview.

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

volosied committed:
Create websocket server API bundle: io.openliberty.jakarta.websocket.2.1

volosied committed:
Create websocket client API bundle: io.openliberty.jakarta.websocket.client.2.1

volosied committed:
Repeat existing websocket tests under EE10

volosied committed:
Create new WebSocket 2.1 FAT: io.openliberty.wsoc.2.1.internal_fat

volosied committed:
Update session timeout logging; cnf dependencies, and websocket 2.1 private feature file

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

pnicolucci commented on dev/io.openliberty.wsoc.21.internal_fat/build.gradle:
```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+* Copyright (c) 2020, 2021 IBM Corporation and others.
+* All rights reserved. This program and the accompanying materials
```
Should this just be 2022 for a new FAT?

volosied replied:
It should. I'll fix it

pnicolucci commented on dev/io.openliberty.jakarta.websocket.2.1/build.gradle:
```diff
@@ -0,0 +1,41 @@
+
+dependencies {
+     sourceJar 'jakarta.websocket:jakarta.websocket-api:2.1.0:sources'
+}
```
Do we normally pull down sources for the API? Is this necessary? I just want us to be consistent when possible.

volosied replied:
We do (at least I have). I see API sources added for MyFaces, EL, and I've done this for Jakarta Tags.  However, I don't see anything for servlet? We can add that in later for consistency.

volosied replied:
Faces-3.0 has the javadocs as well.  Not sure if that necessary, but I think the sources are enough for the APIs

pnicolucci commented:
#libby

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.21.internal_fat

pnicolucci commented:
Need to look at: https://github.com/OpenLiberty/open-liberty/tree/integration/dev/io.openliberty.jakarta there is some duplication between this PR and what was already delivered.

volosied committed:
Remove websocket.2.1.bnd from io.openliberty.jakarta

volosied commented:
I removed [websocket.2.1.bnd from io.openliberty.jakarta](https://github.com/OpenLiberty/open-liberty/pull/21099/commits/397bceecbd2df86d65261a5ce1b3bfc6f8bb2d51)

And renamed the wsoc fat to  use a . between 21 : [io.openliberty.wsoc.2.1.internal_fat](https://github.com/OpenLiberty/open-liberty/pull/21099/commits/64901ea3a936ae4bb24326d5ce26d72775c97680)

volosied commented on dev/io.openliberty.wsoc.2.1.internal_fat/.project:
```diff
@@ -0,0 +1,34 @@
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1657035397348</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
```
Added by VS code -- needs to be removed

volosied commented on dev/io.openliberty.jakarta.websocket.client.2.1/bnd.bnd:
```diff
@@ -0,0 +1,34 @@
+
```
remove new line

volosied commented:
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat,io.openliberty.wsoc.2.1.internal_fat

pnicolucci commented:
#libby
